src/plugins/multiColumnSorting/multiColumnSorting.js

```diff
@@ -0,0 +1,804 @@
+   *
+   * @private
+   */
+  loadOrSortBySettings() {
+    this.columnMetaCache.clear();
+
+    const storedAllSortSettings = this.getAllSavedSortSettings();
+
+    if (isObject(storedAllSortSettings)) {
+      this.sortBySettings(storedAllSortSettings);
+
+    } else {
+      const allSortSettings = this.hot.getSettings().multiColumnSorting;
+
+      this.sortBySettings(allSortSettings);
+    }
+  }
+
+  /**
+   * Sort the table by provided configuration.
+   *
+   * @private
+   * @param {Object} allSortSettings All sort config settings. Object may contain `initialConfig`, `indicator`,
+   * `sortEmptyCells`, `headerAction` and `compareFunctionFactory` properties.
+   */
+  sortBySettings(allSortSettings) {
+    if (isObject(allSortSettings)) {
+      this.columnStatesManager.updateAllColumnsProperties(allSortSettings);
+
+      const initialConfig = allSortSettings.initialConfig;
+
+      if (Array.isArray(initialConfig)) {
+        this.sort(initialConfig);
+      }
+
+    } else {
+      // Extra render for headers. Their width may change.
+      this.hot.render();
+    }
+  }
+
+  /**
+   * Callback for the `beforeChange` hook.
+   *
+   * @private
+   * @param {Array} changes Array of changes.
+   */
+  onBeforeChange(changes) {
+    if (changes === null) {
+      return;
+    }
+
+    // Clear sort only when any cell in already sorted column was changed.
+    arrayEach(changes, ([, prop, oldVal, newVal]) => {
+      const visualColumn = this.hot.propToCol(prop);
+      const physicalColumn = this.hot.toPhysicalColumn(visualColumn);
+
+      if (this.columnStatesManager.isColumnSorted(physicalColumn) && oldVal !== newVal) {
+        this.removeSortAction();
+
+        return false;
+      }
+    });
+  }
+
+  /**
+   * Callback for the `afterCreateRow` hook.
+   *
+   * @private
+   * @param {Number} index Visual index of the created row.
+   * @param {Number} amount Amount of created rows.
+   */
+  onAfterCreateRow(index, amount) {
+    this.rowsMapper.shiftItems(index, amount);
+
+    this.removeSortAction();
+  }
+
+  /**
+   * Callback for the `afterRemoveRow` hook.
+   *
+   * @private
+   * @param {Number} removedRows Visual indexes of the removed row.
+   * @param {Number} amount  Amount of removed rows.
+   */
+  onAfterRemoveRow(removedRows, amount) {
+    this.rowsMapper.unshiftItems(removedRows, amount);
+
+    this.removeSortAction();
+  }
+
+  /**
+   * Indicates if clickable header was clicked.
+   *
+   * @param {MouseEvent} event The `mousedown` event.
+   * @param {Number} column Visual column index.
+   * @returns {Boolean}
+   */
+  wasClickableHeaderClicked(event, column) {
+    const pluginSettingsForColumn = this.getFirstCellSettings(column).multiColumnSorting;
+    const headerActionEnabled = pluginSettingsForColumn.headerAction;
+
+    if (headerActionEnabled && event.realTarget.nodeName === 'SPAN') {
+      return true;
+    }
+
+    return false;
```

Comment:
You can just return: `return headerActionEnabled && event.realTarget.nodeName === 'SPAN';`